src/Consumer.php

```diff
@@ -139,7 +139,13 @@ function (AMQPMessage $message) use ($connection, $options, $connectionName, $qu
             // Finally, we will check to see if we have exceeded our memory limits or if
             // the queue should restart based on other indications. If so, we'll stop
             // this worker and let whatever is "monitoring" it restart the process.
-            $this->stopIfNecessary($options, $lastRestart, $this->gotJob ? true : null);
+            $status = $this->stopIfNecessary(
+                $options, $lastRestart, $startTime, $jobsProcessed, $job
+            );
+
+            if (! is_null($status)) {
+                return $this->stop($status);
```

Comment:
A void function must not return a value

Reply:
ah, return is invalid. See parent class from laravel https://github.com/laravel/framework/blob/bc7d4b7fb1a444c1646d7d47a75185b72e78537d/src/Illuminate/Queue/Worker.php#L117